src/server/test/pegasus_write_service_impl_test.cpp

```diff
@@ -140,5 +156,66 @@ TEST_F(pegasus_write_service_impl_test, verify_timetag_compatible_with_version_0
     dsn::fail::teardown();
 }
 
+TEST_F(pegasus_write_service_impl_test, incr_on_absent_record)
+{
+    dsn::apps::incr_request req;
+    req.increment = 100;
+    pegasus::pegasus_generate_key(
+        req.key, dsn::string_view("hash_key"), dsn::string_view("sort_key"));
+
+    db_get_context get_ctx;
+    db_get(req.key, &get_ctx);
+    ASSERT_FALSE(get_ctx.found);
+
+    dsn::apps::incr_response resp;
+    _write_impl->incr(0, req, resp);
+    ASSERT_EQ(resp.new_value, 100);
+
+    db_get(req.key, &get_ctx);
+    ASSERT_TRUE(get_ctx.found);
+}
+
+TEST_F(pegasus_write_service_impl_test, negative_incr_and_zero_incr)
+{
+    dsn::apps::incr_request req;
+    req.increment = -100;
```

Comment:
there are too many same code, may be you can refactor it, such as using for look to do it.

Reply:
Take a look at the new code. I've refactored much. I think it is compacted enough.